target/linux/lantiq/dts/RS353.dts

```diff
@@ -0,0 +1,318 @@
+/dts-v1/;
+
+/include/ "vr9.dtsi"
+
+/ {
+	model = "RS353 - Bintec Elmeg RS353";
+
+	chosen {
+		bootargs = "console=ttyLTQ0,115200 init=/etc/preinit";
+		leds {
+                        running = &status;
+                        boot = &status;
+			wifi = &wlan;
+			usb2 = &usb; /* Front port is mapped to device 2 */
+			dsl = &dsl;
+                };
+
+	};
+
+	memory@0 {
+		reg = <0x0 0x8000000>;
+	};
+
+	sram@1F000000 {
+		cgu0: cgu@103000 {
+                        compatible = "lantiq,cgu-xway";
+                        reg = <0x103000 0x1000>;
+                        #clock-cells = <1>;
+			lantiq,phy-clk-src = <0x2>;
+		};
+	};
+
+	fpi@10000000 {
+		localbus@0 {
+			nor-boot@0 {
+				compatible = "lantiq,nor";
+				bank-width = <2>;
+				reg = <0 0x0 0x2000000>;
+				#address-cells = <1>;
+				#size-cells = <1>;
+
+				partitions {
+					compatible = "fixed-partitions";
+					#address-cells = <1>;
+					#size-cells = <1>;
+
+					boardconfig: partition@0 {
```

Comment:
I was able to get some background info regarding the EBU. It might be a bit hard to read, since I got this informations in a chat and not in english. Might be that I got some things wrong.

The EBU is an 16 bit intel bus which is always little endian. But the CPU is big endian. Therefore the endianess swap is required. By the way, I was told that his bit is related to PCI swapping as well. 

The lantiq SoCs have an internal address swapping unit. But all lantiq SoCs up to xrx200 v1.1 (including amazon, amazonSE, danube) have the bug that swapping is only done in write direction. This bug was fixed with vr9 v1.2 and doesn't exists in xrx300+. Because of that, the in SoC swapping is not enabled by most bootloaders and the swapping is done in software. Back in the days, lantiq simply stored endianess swabbed data on the flash to bypass this bug.

But you have a xRX200 rev 1.2, your bootmonitor bootloader knows about it and enables the in SoC swapping. But the mtd driver still expects that he has to do all the swapping.

Bit 30 of the EBU_CON_0 register is the address swapping bit.  0 = address swapping unit disabled (default) and 1 = address swapping unit enabled.

I had a chat with the lantiq-flash.c author and he had confirmed that doing the in driver swap based on the bit should work and shouldn't break existing boards. Not sure if it would even be better to force the write swab if the address swapping unit is enabled (bit set) but it's an SoC < xRX200 v1.2.

At least at LEDE we are trying hard to get rid of the OpenWrt/LEDE specific kernel patches by upstreaming them. I strongly suggest to send this patch first to the linux-mtd and cc to linux-mips mailinglist to get your patch into the kernel. After it was accepted by the kernel people backport the patch to openWrt/LEDE.

Will reply to the other topics later.